ef_tx.go

```diff
@@ -0,0 +1,68 @@
+package bux
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/libsv/go-bt/v2"
+)
+
+func ToEfHex(ctx context.Context, tx *Transaction, store TransactionGetter) (efHex string, ok bool) {
+	btTx := tx.parsedTx
+
+	if btTx == nil {
+		var err error
+		btTx, err = bt.NewTxFromString(tx.Hex)
+		if err != nil {
+			return "", false
+		}
+	}
+
+	needToHydrate := false
+	for _, input := range btTx.Inputs {
+		if input.PreviousTxScript == nil {
+			needToHydrate = true
+			break
+		}
+	}
+
+	if needToHydrate {
+		if ok := hydrate(ctx, btTx, store); !ok {
+			return "", false
+		}
+	}
+
+	return hex.EncodeToString(btTx.ExtendedBytes()), true
+}
+
+func hydrate(ctx context.Context, tx *bt.Tx, store TransactionGetter) (ok bool) {
+	txToGet := make([]string, 0, len(tx.Inputs))
+
+	for _, input := range tx.Inputs {
+		txToGet = append(txToGet, input.PreviousTxIDStr())
+	}
+
+	parentTxs, err := store.GetTransactionsByIDs(ctx, txToGet)
+	if err != nil {
+		return false
+	}
```

Comment:
By returning `boolean` value instead of `error` from the `hydrate` func, we lose the information what really happened. For example we're losing the error content here.

Maybe it is not necessary to log these events - if so, just resolve the comment.

Reply:
I did intentionally, I think we don't need error info here - just simple success or not